### PR TITLE
OOP Classes And Objects / it-academy.by / JC1-110-22 / Maksim Bondar / Tasks 1 - 4 complete

### PR DESCRIPTION
**Pull request by shiv-qa-auto**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+public class Time {
+
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private int totalSeconds;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:8]: Remove filed which you can operate

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+public class Atm {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, plese review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public int getNumberBanknotes50() {
+        return this.NumberBanknotes50;
+    }
+
+    public int getNumberBanknotes100() {
+        return this.NumberBanknotes100;
+    }
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, plese review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+public class Point {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:3]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+public class Rectangle {
+
+    Point topLeftPoint;
+    Point bottomRightPoint;
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public Rectangle (Point topLeftPoint, Point bottomRightPoint) {
+        this.topLeftPoint = topLeftPoint;
+        this.bottomRightPoint = bottomRightPoint;
+    }
+
+    public Point getTopLeftPoint() {
+        return this.topLeftPoint;
+    }
+
+    public void setTopLeftPoint(int x, int y) {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+
+public class Time {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:3]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java (hunk 1 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    private int numberBanknotes20;
+    private int NumberBanknotes50;
+    private int NumberBanknotes100;
+
+    public Atm (int numberBanknotes20, int numberBanknotes50, int numberBanknotes100) {
+        this.numberBanknotes20 = numberBanknotes20;
+        this.NumberBanknotes50 = numberBanknotes50;
+        this.NumberBanknotes100 = numberBanknotes100;
+    }
+    public void addBanknotes20 (int x) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:12]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        this.numberBanknotes20 += x;
+    }
+    public void addBanknotes50 (int x) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:15]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        this.NumberBanknotes50 += x;
+    }
+    public void addBanknotes100 (int x) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:18]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        this.NumberBanknotes100 += x;
+    }
+    public int getNumberBanknotes20() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:21]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        return this.numberBanknotes20;
+    }
+    public int getNumberBanknotes50() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:24]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        return this.NumberBanknotes50;
+    }
+    public int getNumberBanknotes100() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java:27]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+public class Point {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:3]:
```suggestion
package com.rakovets.course.java.core.practice.oop_classes_and_objects;

public class Point {
```

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    private int x;
+    private int y;
+
+    public Point (int x, int y) {
+        this.x = x;
+        this.y = y;
+    }
+    public int getX() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:11]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        return x;
+    }
+    public void setX(int x) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:14]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+        this.x = x;
+    }
+    public int getY() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:17]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+        return y;
+    }
+    public void setY(int y) {
+        this.y = y;
+    }
+    public double getDistance(Point p) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java:23]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 19]
```diff
@@ -0,0 +1,31 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+public class Rectangle {
+
+    private Point topLeftPoint;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java:4]:
```suggestion
package com.rakovets.course.java.core.practice.oop_classes_and_objects;

public class Rectangle {
    private Point topLeftPoint;
```

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+    public int getMinutes() {
+        return this.minutes;
+    }
+    public void setMinutes(int minutes) {
+        this.minutes = minutes;
+    }
+    public int getSeconds() {
+        return this.seconds;
+    }
+    public void setSeconds(int seconds) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:35]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+        this.seconds = seconds;
+    }
+    public int getTotalSeconds() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java:38]: "Violation: There isn't separating empty line
Solution: Add blank line"

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 8 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 10 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 11 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 12 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java (hunk 13 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java (hunk 15 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java (hunk 17 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+    public int getX() {
+        return x;
+    }
+    public void setX(int x) {
+        this.x = x;
+    }
+    public int getY() {
+        return y;
+    }
+    public void setY(int y) {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java (hunk 18 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Point.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java (hunk 19 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+    private Point bottomRightPoint;
+
+    public Rectangle (Point topLeftPoint, Point bottomRightPoint) {
+        this.topLeftPoint = topLeftPoint;
+        this.bottomRightPoint = bottomRightPoint;
+    }
+    public Point getTopLeftPoint() {
+        return this.topLeftPoint;
+    }
+    public void setTopLeftPoint(Point topLeftPoint) {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+        this.topLeftPoint = topLeftPoint;
+    }
+    public Point getBottomRightPoint() {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+        return this.bottomRightPoint;
+    }
+    public void setBottomRightPoint(Point bottomRightPoint) {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+        this.bottomRightPoint = bottomRightPoint;
+    }
+    public int getPerimeter() {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Rectangle.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 27]
```diff
@@ -0,0 +1,42 @@
+package com.rakovets.course.java.core.practice.oop_classes_and_objects;
+public class Time {
+
+    private int hours;
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+        this.hours = hours;
+        this.minutes = minutes;
+        this.seconds = seconds;
+    }
+    public Time (int totalSeconds) {
+        this.hours = totalSeconds / SECONDS_IN_HOUR;
+        this.minutes = totalSeconds % SECONDS_IN_HOUR / SECONDS_IN_MINUTE;
+        this.seconds = totalSeconds % SECONDS_IN_HOUR % SECONDS_IN_MINUTE;
+    }
+    public int getHours() {
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java (hunk 20 above)
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Time.java]: Fixed, please review

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**